src/data_factory_testing_framework/functions/evaluator/rules/evaluation_expression_rule.py

```diff
@@ -0,0 +1,40 @@
+from lark import Tree
+
+from data_factory_testing_framework.exceptions.expression_evaluation_error import ExpressionEvaluationError
+from data_factory_testing_framework.functions.evaluator.exceptions import ExpressionEvaluationInvalidChildTypeError
+from data_factory_testing_framework.functions.evaluator.rules.expression_rule import EvaluatedExpression
+
+from .expression_rule import ExpressionRuleEvaluator
+
+
+class EvaluationExpressionRuleEvaluator(ExpressionRuleEvaluator):
+    def __init__(self, tree: Tree) -> None:
+        """Initializes the expression rule evaluator."""
+        super().__init__(tree)
+
+        if len(self.children) < 1:
+            raise ExpressionEvaluationError(
+                f"Invalid number of children. Minimum required: 1, Actual: {len(self.children)}"
+            )
+
+        if not isinstance(self.children[0], ExpressionRuleEvaluator):
+            raise ExpressionEvaluationInvalidChildTypeError(
+                child_index=0, expected_types=ExpressionRuleEvaluator, actual_type=type(self.children[0])
+            )
+
+    def evaluate(self) -> EvaluatedExpression:
+        expression = self.children[0]
```

Comment:
Maybe we should move the unpacking of children to the constructor and also do all the type checks there, so that the evaluate method is purely about evaluating and the constructor about constructing and validating?

Reply:
Good idea actually!